packages/infra/lib/api-stack.ts

```diff
@@ -436,6 +433,21 @@ export class APIStack extends Stack {
     const oauthAuth = this.setupOAuthAuthorizer(userPoolClientSecret);
     this.setupAPIGWOAuthResource(id, api, link, oauthAuth, oauthScopes, apiLoadBalancerAddress);
 
+    const contributionResource = api.root.addResource("doc-contribution");
+
+    // setup cw doc contribution
+    const cwDocContributionLambda = this.setupCWDocContribution({
+      baseResource: contributionResource,
+      lambdaLayers,
+      alarmAction: slackNotification?.alarmAction,
+      authorizer: oauthAuth,
+      oauthScopes: oauthScopes,
+      envType: props.config.environmentType,
+      bucketName: props.config.medicalDocumentsBucketName,
+    });
+
+    medicalDocumentsBucket.grantReadWrite(cwDocContributionLambda);
```

Comment:
If we pass the `medicalDocumentsBucket` as a param to `setupCWDocContribution` we can get the bucket name and provide access to the lambda all within the function.